packages/backend/src/auth-service-client/client.ts

```diff
@@ -0,0 +1,88 @@
+interface Tenant {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+export class AuthServiceClientError extends Error {
+  constructor(
+    message: string,
+    public status: number,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public details?: any
+  ) {
+    super(message)
+    this.status = status
+    this.details = details
+  }
+}
+
+export class AuthServiceClient {
+  private baseUrl: string
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async request<T = any>(
+    path: string,
+    options: RequestInit
+  ): Promise<T> {
+    const response = await fetch(`${this.baseUrl}${path}`, options)
+
+    if (!response.ok) {
+      let errorDetails
+      try {
+        errorDetails = await response.json()
+      } catch {
+        errorDetails = { message: response.statusText }
+      }
+
+      throw new AuthServiceClientError(
+        `Auth Service Client Error: ${response.status} ${response.statusText}`,
+        response.status,
+        errorDetails
+      )
+    }
+
+    if (
+      response.status === 204 ||
+      response.headers.get('Content-Length') === '0'
+    ) {
+      return undefined as T
+    }
+
+    const contentType = response.headers.get('Content-Type')
+    if (contentType && contentType.includes('application/json')) {
+      try {
+        return (await response.json()) as T
+      } catch (error) {
+        throw new AuthServiceClientError(
+          `Failed to parse JSON response from ${path}`,
+          response.status
+        )
+      }
+    }
+
+    return (await response.text()) as T
+  }
```

Comment:
I opted to use the native `fetch` here although there were some behaviors I wasnt quite sure about. I tested with `axios` and mirrored how that works:

1. not OK response (400, 500) etc. throws
2. 204 returns `undefined`
3. Success without body (200, 201, etc.) returns the response.text (just `''` if not set)
4. Interface accepts a generic which asserts the return type, although technically it could still return a `string` or `undefined` in the cases described above.